doc/source/getting_started/WorkingWithInferenceData.ipynb

```diff
@@ -13,11 +13,12 @@
   },
```

Comment:
<p>I like the idea of using the variable just created for a 2nd computation. The cumulative mean however I think is to complicated for this tutorial is a bit too complicated for this tutorial. It is also using <code>len</code>  and positional indexing to loop over chains which we should try to avoid. Maybe there is something else we could do instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/arviz-devs/arviz/pull/1697/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Reply:
I used xarray rolling mean instead, have a look and let me know if that works? also, just curious, are len and positional indexing considered bad practices?

Reply:
I would say they are borderline bad practices because xarray code should ideally not depend on the order of the dimensions, only on their name. Therefore, doing `len(dataarray)` won't do this, the ideal approach would be `len(dataarray.dim_name)` so we are explicit on which is the dim whose length we want instead of grabbing the first dimension. 

Currently ArviZ enforces the `chain, draw, *shape` convention by reshaping and adding indices but this is an _input_ convention and code should not take as a property that once converted to xarray this order will still hold, because it will eventually break due to transpositions or https://github.com/arviz-devs/arviz/issues/1693. 

The convention should be understood as "if ArviZ gets unlabeled arrays as input it will default to interpreting their dimensions as `chain, draw, *shape`" not as something applied to the data once converted to labeled arrays